### PR TITLE
Removed loop from bomb relocation code

### PR DESCRIPTION
**Pull request by EoinDoherty**
engine.py's _check_safe_first_move method now randomly chooses new locations from the set of tiles on the board that do not contain a bomb. 

I also changed default -ai parameter so the else statement in game.py is reachable.

### CODE REVIEW

File: engine/engine.py [hunk 1]
```diff
@@ -115,14 +115,29 @@ def _unsafe_check_neighbors(self, x, y):
...
-            new_coords = np.floor(np.multiply(np.random.rand(nb, 2), np.array([[self._width, self._height]]))).astype(int)
-            for nc in new_coords:
-                while self._board[nc[0], nc[1]] == Spaces.BOMB:
-                    nc = np.floor(np.multiply(np.random.rand(1, 2), np.array([[self._width, self._height]]))).astype(int)[0]
-                self._board[nc[0], nc[1]] = Spaces.BOMB
-            nb = self._get_neighboring_bombs(x, y)
+
+            empty_tiles = []
+
+            for x in range(self._width):
```
Comment (KyleRosenberg) [inline engine/engine.py]: I hate to say it but could you add a unit test to make sure the number of bombs is correct after this runs?

**Review by KyleRosenberg — CHANGES_REQUESTED** [review]

File: engine/engine.py (hunk 1 above)
Comment (KyleRosenberg) [reply engine/engine.py]: It seems like with this logic, the original space could be picked as a bomb again since you cleared the neighbors beforehand

Comment (EoinDoherty) [reply engine/engine.py]: Oh right duh.

Comment (EoinDoherty) [reply engine/engine.py]: That should be fixed now

**Review by KyleRosenberg — APPROVED** [review]

**Pull request merged**